utils/docker-utils.sh

```diff
@@ -129,7 +138,87 @@ docker::prune_configs() {
 
     # shellcheck disable=SC2046
-    docker config rm $(docker config ls -qf label=name="$CONFIG_LABEL") &>/dev/null
+    if [[ -n $(docker config ls -qf label=name="$CONFIG_LABEL") ]]; then
+        log info "Waiting for configs to be removed..."
+
+        docker config rm $(docker config ls -qf label=name="$CONFIG_LABEL") &>/dev/null
+
+        overwrite "Waiting for configs to be removed... Done"
+    fi
+}
+
+# Deploy a service, it will set config digests (in case a config is defined in the compose file)
+#
+# Arguments:
+# - $1 : docker compose path, e.g. "/instant/monitoring" ...
+# - $2 : docker compose file, e.g. "docker-compose.yml" "docker-compose.cluster.yml" ...
+# - $3 : docker compose dev file, e.g. "docker-compose.dev.yml"
+# - $4 : services names, e.g. "monitoring" "hapi-fhir" ...
+#
+docker::deploy_service() {
+    local -r DOCKER_COMPOSE_PATH="${1:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_FILE="${2:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_DEV_FILE="${3:-""}"
+    local docker_compose_dev=""
+
+    # Check for need to set config digests
+    local -r files=($(yq '.configs."*.*".file' "${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE"))
+    if [ "${files[*]}" != "null" ]; then
+        config::set_config_digests "${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE"
+    fi
+
+    if [ -n "${DOCKER_COMPOSE_DEV_FILE}" ]; then
+        docker_compose_dev="-c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_FILE"
+    fi
+
+    try "docker stack deploy \
+        -c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE \
+        $docker_compose_dev \
+         instant" \
+        throw \
+        "Wrong configuration in compose file"
+}
+
+# Deploy a config importer:
+# Sets the config digests, deploy the config importer, remove it and remove the stale configs
+#
+# Arguments:
+# - $1 : docker compose path, e.g. "/instant/monitoring/importer/docker-compose.config.yml" ...
+# - $2 : services name, e.g. "clickhouse-config-importer" ...
+# - $3 : config label, e.g. "clickhouse" "kibana" ...
+# - $4 : docker compose file name, default "docker-compose.config.yml"
+#
+docker::deploy_config_importer() {
+    local -r CONFIG_COMPOSE_PATH="${1:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r SERVICE_NAME="${2:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r CONFIG_LABEL="${3:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+
+    log info "Waiting for config importer $SERVICE_NAME to start ..."
+    (
+        if [ ! -f "$CONFIG_COMPOSE_PATH" ]; then
+            log error "No such file: $CONFIG_COMPOSE_PATH"
+            exit 1
+        fi
+
+        config::set_config_digests "$CONFIG_COMPOSE_PATH"
+
+        try \
+            "docker stack deploy -c ${CONFIG_COMPOSE_PATH} instant" \
+            throw \
+            "Wrong configuration in compose file"
```

Comment:
Reference to the compose file with errors here would be nice, if a service has two config importers you would then know which one of them failed.

Reply:
cool yes, fixed in [4d4b85c](https://github.com/jembi/platform/pull/188/commits/4d4b85c17604540e8d31869e0e0bc75f05133470)